src/celeritas/global/Stepper.cc

```diff
@@ -0,0 +1,155 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2022 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/global/Stepper.cc
+//---------------------------------------------------------------------------//
+#include "Stepper.hh"
+
+#include "corecel/Assert.hh"
+#include "corecel/data/Ref.hh"
+#include "celeritas/phys/PhysicsParams.hh"
+#include "celeritas/phys/Primary.hh"
+#include "celeritas/track/TrackInitParams.hh"
+#include "celeritas/track/TrackInitUtils.hh"
+
+#include "ActionManager.hh"
+#include "CoreParams.hh"
+
+namespace celeritas
+{
+namespace
+{
+//---------------------------------------------------------------------------//
+std::vector<ActionId> build_action_vec(const CoreParams& core)
+{
+    std::vector<ActionId> actions;
+
+    // TODO: add accessors instead of looking these up as strings
+    const ActionManager& action_mgr = *core.action_mgr();
+    for (const char* action_label : {
+             "pre-step",
+             "along-step",
+             "physics-discrete-select",
+             "geo-boundary",
```

Comment:
It shouldn't: the only dependency on the graph of actions is that both of them happen before the create-from-secondaries kernel:
![actions](https://user-images.githubusercontent.com/741229/167834276-27575b78-74d1-476e-a2f4-94128e075b0d.png)

Reply:
Ohh nice graph. Makes sense, thanks!

Reply:
Indeed, a nice graph!  A couple of comments: 1) better to break this graph for electrons (charged) and photon (neutral), especially that photons do not follow  `range limited, along-step, MSC range` box at all - at least additional color code for photon may helpful if both put in the same graph.  2) along-step box for charged particles may be enlarged and contains sub-boxes for `MSC-range, field propagator and MSC-scattering, dedx calculation` of which ordering matters (even though all their actions are inside a single big action).  For photon, the boundary action, in some sense, may also cover the Along-step action (i.e., linear propagator).  Details may not be important or add more complexity, but may avoid a potential confusion so that this graph can serve as the whole celeritas action flow.

Reply:
No sweat at all for now, even though **no energy loss process** is logically equivalent to **neutral particle process** or vice versa (i.,e these two are completely inter-changeable without an exception, I guess). For the sake of further note or later tests, we may also try/support transporting photons independently/separately from e+/e- even though it will require different shuffling for `extend from secondaries` at the end of each-transport loop which involves extra cross talks or sorting (which may be expensive) between two kernels for preparing same kind of tracks for the next loop. So, the question is what are trades-off between these alternative approaches, especially executing kernels for neutrals separately which skip most of heaviest tasks (msc, field, dedx), but along with a (smarter?) high level kernel scheduling - which may be also a good R&D with multiple GPUs.

Reply:
Yep, and don't forget a third option is going to be sorting tracks by particle type before the heavy kernels (like along-step), so there will at least be less divergence within warps...